app/rest/model/src/main/java/io/syndesis/model/integration/IntegrationRevision.java

```diff
@@ -16,61 +16,121 @@
 package io.syndesis.model.integration;
 
 import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
-import org.immutables.value.Value;
+import io.syndesis.core.SuppressFBWarnings;
+import io.syndesis.model.Kind;
+import io.syndesis.model.WithId;
+import io.syndesis.model.WithKind;
+import io.syndesis.model.WithName;
 
 import java.math.BigInteger;
+import java.util.Collections;
 import java.util.Date;
+import java.util.List;
 import java.util.Optional;
 
-@Value.Immutable
 @JsonDeserialize(builder = IntegrationRevision.Builder.class)
-public abstract class IntegrationRevision {
+public class IntegrationRevision implements WithKind, WithId<IntegrationRevision>, WithName {
```

Comment:
Shouldn't this class be renamed to IntegrationDeployment ?

Reply:
I find `IntegrationDeployment` a little bit confusing / misleading. Don't have a strong objection though, so I can rename it if you insist :-)

Reply:
Revision gives the impression that it gets created every time you update the integration when in fact we create these when we try to deploy them.